### PR TITLE
feat: support disablement of APM Tracing (the product)

### PR DESCRIPTION
**Pull request by dmehala**
# Description
Supersede #217 

# Motivation
1. **Simplify the Implementation**
While the original approach worked, I identified a more straightforward way to reach the same goals with fewer moving parts and a clearer mental model.
2. **Clarify Underspecified Behavior**
Some behaviors, especially around context extraction, aren’t fully defined in the RFC _yet_. Rather than making assumptions, we took the opportunity to clarify and solidify these behaviors.
3. **Improve Long-Term Maintainability**
Reusing existing constructs and patterns makes the codebase easier to understand, test, and evolve. This rework brings the implementation closer to that goal.

## Improvements (non-exhaustive)
* **Avoided injecting empty tracing context**, which required adding support to delete headers on the integration side.
* **Added more comprehensive tests** to cover edge cases.
* **Aligned API design for trace source attribution**, moved from injection options to span-level attributes for clearer semantics.
* **Simplified sampling logic** by using existing rule-based mechanisms rather than introducing new types or abstractions.
* **Clarified the API for controlling stats computation**, improving long term maintainability.
* **Validated extracted trace source** to ensure consistency and reduce potential for errors.
* **Reused existing concepts and components**, avoiding the need for new structures and reducing the risk of regressions.

### CODE REVIEW

**Review by cataphract — APPROVED** [review]

File: src/datadog/trace_source.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  auto maybe_ts_uint = parse_uint64(source_str, 10);
+  if (maybe_ts_uint.if_error()) return false;
+
+  // Bit twiddling magic is coming from
+  // <http://www.graphics.stanford.edu/~seander/bithacks.html> <3.
+  auto is_power_of_2 = [](uint64_t v) -> bool { return v && !(v & (v - 1)); };
+
+  return is_power_of_2(*maybe_ts_uint);
+}
```
[the remark this answers is not in the document]
Comment (dmehala) [reply src/datadog/trace_source.cpp]: Good catch, thank you. Fixed in ff2c7eab8af1afb14018101430f3156d068fe9eb
